### ISSUE TITLE
[Guidance Request] Best Practices for Reconstructing Partial-Band (10-Band) Sentinel-2 Data

### ISSUE THREAD

**Issue by Luo-Z13** [IBM/terramind#36]
Hi TerraMind team,

First of all, thank you for developing and open-sourcing this incredibly powerful foundational model for remote sensing. It's a fantastic contribution to the community.

I'm currently exploring the use of the pre-trained `terramind_v1_tokenizer_s2l2a` VAE for reconstructing Sentinel-2 L2A imagery, specifically on datasets like [PASTIS-R](https://zenodo.org/records/5735646) which provide 10 out of the 12 standard S2L2A bands (**missing B1/COASTAL_AEROSOL and B9/WATER_VAPOR**).

My preprocessing pipeline was as follows:

1. Take the 10-band image.
2. Apply zero-padding at the positions of B1 (index 0) and B9 (index 9) to create a 12-band tensor.
3. Normalize this 12-band tensor using the official `v1_pretraining_mean` and std.
4. Encode it into the latent space and then decode it back.
5. The reconstruction results were unfortunately very poor, with significant color distortion and loss of detail, as shown below (using two different samples from PASTIS-R).

<img width="1994" height="710" alt="Image" src="https://github.com/user-attachments/assets/78c6fc20-2b59-48f4-94fc-f4f7b3227f57" />

<img width="1996" height="716" alt="Image" src="https://github.com/user-attachments/assets/d29d3c34-cc00-4911-ad8e-c5552e6ee8ba" />

To diagnose whether the zero-padding itself was the issue, I conducted a follow-up experiment on a full 12-band L2A image from your examples. I compared the reconstruction of the original image with the reconstruction of a "corrupted" version where I manually set the B1 and B9 bands to zero after normalization.

<img width="4200" height="1200" alt="Image" src="https://github.com/user-attachments/assets/648da9bb-14c2-4254-b0bc-e3e4406e97c4" />

<img width="4200" height="1200" alt="Image" src="https://github.com/user-attachments/assets/8039ec97-9b07-40a6-a8c1-77dab2652bb0" />

As you can see:

- The VAE perfectly reconstructs the original, uncorrupted 12-band image (Column 2).
- However, when B1 and B9 are set to zero, the reconstruction quality degrades dramatically (Columns 3 & 4), exhibiting the same kind of color shift and blurring seen with the PASTIS-R data.

This strongly suggests that the VAE is sensitive to this zero-padding strategy, likely treating it as a significant out-of-distribution input.

**My question is**: Do you have any recommended best practices for using the S2L2A VAE on datasets with missing bands? Is there a better strategy than zero-padding, or would fine-tuning the VAE on this padded data be the only viable path forward?

Any insights would be greatly appreciated. Thank you for your time!

**Comment by Luo-Z13** [comment]
> Hi [@Luo-Z13](https://github.com/Luo-Z13), as you pointed out, dropping based is OOD for the model, so I don't think you can use it without any additional fine-tuning. The only way that comes to my mind and might help is to do padding with the mean channel value. This is similar to dropping the band in the patch embedding. This is how it is also done in TerraTorch for TerraMind and Prithivi if you use a subset of bands for fine-tuning.
> 
> But ideally you would integrate the tokeniser in your training pipeline and do some continous training (with the bands dropped in the patch embedding so you don't need to pad the data, see https://github.com/terrastackai/terratorch/blob/1beb67d81bb92917e8e9a27ab036f934f97c747c/terratorch/models/backbones/terramind/model/terramind_register.py#L214)

Thank you so much for the incredibly insightful response and guidance! This is immensely helpful.
